src/verify.c

```diff
@@ -805,6 +809,38 @@ int verify_main(int argc, char **argv)
 	}
 	grabtime_stop(&times);
 
+	/* When installing, check disk space availability */
+	/* This assumes there's no pre-installed/partially installed bundles in path_prefix */
+	if (cmdline_option_install) {
+		struct list *ptr = list_head(official_manifest->submanifests);
+		while (ptr) {
+			struct manifest *subman;
+			subman = ptr->data;
+			ptr = ptr->next;
+
+			total_size += subman->contentsize;
+		}
+
+		/* Calculate free space on filepath */
+		if (statvfs(path_prefix, &stat) != 0) {
+			fs_free = -1;
+		} else {
+			fs_free = stat.f_bsize * stat.f_bavail;
+		}
+
+		/* Add 10% to total_size as a 'fudge factor' */
+		if (((total_size * 1.1) > fs_free) || fs_free < 0) {
+			ret = EBUNDLE_INSTALL;
+
+			if (fs_free >= 0) {
+				fprintf(stderr, "Error: Files to install too large by %ldM.\n", (total_size - fs_free) / 1000 / 1000);
+			} else if (fs_free < 0) {
+				fprintf(stderr, "Error: Unable to determine free space on filesystem.\n");
+			}
+			goto clean_and_exit;
```

Comment:
Didn't we discuss not failing for the install case? We don't want our installer frontends to fail out with no recourse for the user, especially when this is potentially a liberal check in case os-core is already installed on the target for some reason.

Reply:
I should include some way to skip the check in case the user wants to try anyways.

In issue #328, the requested behavior was to print an error and exit.

I think the next step is to change do_staging to report out of disk errors - but without a pre-check, we will have to keep going (because we're after the "point of no return"), and print an error/run the failure path for each file in the manifests, which takes a lot of time.

As an example, using verify --install on a full filesystem to verify 8301 missing files takes swupd 93 seconds to fail, while printing ~16000 lines of errors. I think this may be why the bug submitter stated swupd hung, because I couldn't reproduce the hang, just a long processing time.

I'm not sure what to do here... I think we need to speed up the failure case, since we can't exit early. Suggestions?